DynamicProbes.csproj

```
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net8.0</TargetFramework>
+    <RuntimeIdentifier>linux-x64</RuntimeIdentifier>

```

```suggestion
    <RuntimeIdentifier>linux</RuntimeIdentifier>
```

Would just [linux](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/PortableRuntimeIdentifierGraph.json#L300) be more correct?

---

Ah, it gives me `NETSDK1084: There is no application host available for the specified RuntimeIdentifier` on a x64 machine.

But [`RuntimeIdentifiers`](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#runtimeidentifiers) compiles:

```suggestion
    <RuntimeIdentifiers>linux-x64;linux-x86</RuntimeIdentifiers>
```

What do you think about it?

---

Huh, maybe libstapsdt is only for x64: https://github.com/linux-usdt/libstapsdt/issues/33

Then let's stick to linux-x64 for now :)

---

> Huh, maybe libstapsdt is only for x64: https://github.com/linux-usdt/libstapsdt/issues/33

Hah!

> Then let's stick to linux-x64 for now :)

My reason for setting `linux-x64` was keeping [Native AOT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net8plus%2Cwindows#platformarchitecture-restrictions) in mind and that `linux-x86` isn't _officially_ supported.

While [list of runtime identifiers listed in the catalog](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog) is large, the [supported ones for cross-gen can be found in the SDK](https://github.com/dotnet/sdk/blob/d672b8a0459b4888b3e9a6530646e61dd14941b2/src/Installer/redist-installer/targets/GenerateBundledVersions.targets#L320-L353).

See also:

- https://github.com/dotnet/runtime/issues/7335
- https://github.com/dotnet/runtime/issues/86573
- https://github.com/dotnet/runtime/issues/31180